Address resource warnings because of open file handles

rahulporuri opened the pull request:
Saving the graphics context to the PS and SVG backends raises resource warnings because the file handler is not closed by the save method. This PR fixes the issue by using open as a context manager.

For example, the resource warnings look like this on master - 

```
test_circle (test_ps_drawing.TestPSDrawing) ... ResourceWarning: unclosed file <_io.TextIOWrapper name='C:\\Users\\rporuri\\AppData\\Local\\Temp\\tmp5jjp1zn9\\rendered.eps' mode='w' encoding='cp1252'> [test_ps_drawing.py:17]
ok
test_circle_clip (test_ps_drawing.TestPSDrawing) ... ResourceWarning: unclosed file <_io.TextIOWrapper name='C:\\Users\\rporuri\\AppData\\Local\\Temp\\tmpd9y5my5y\\rendered.eps' mode='w' encoding='cp1252'> [test_ps_drawing.py:17]
ok
test_circle_fill (test_ps_drawing.TestPSDrawing) ... ResourceWarning: unclosed file <_io.TextIOWrapper name='C:\\Users\\rporuri\\AppData\\Local\\Temp\\tmpo6l4mdcw\\rendered.eps' mode='w' encoding='cp1252'> [test_ps_drawing.py:17]
ok
test_line (test_ps_drawing.TestPSDrawing) ... ResourceWarning: unclosed file <_io.TextIOWrapper name='C:\\Users\\rporuri\\AppData\\Local\\Temp\\tmpgixdhapv\\rendered.eps' mode='w' encoding='cp1252'> [test_ps_drawing.py:17]
ok
test_quarter_circle (test_ps_drawing.TestPSDrawing) ... ResourceWarning: unclosed file <_io.TextIOWrapper name='C:\\Users\\rporuri\\AppData\\Local\\Temp\\tmpid_gsfje\\rendered.eps' mode='w' encoding='cp1252'> [test_ps_drawing.py:17]
ok
test_rect (test_ps_drawing.TestPSDrawing) ... ResourceWarning: unclosed file <_io.TextIOWrapper name='C:\\Users\\rporuri\\AppData\\Local\\Temp\\tmp6rmirwob\\rendered.eps' mode='w' encoding='cp1252'> [test_ps_drawing.py:17]
ok
test_rectangle (test_ps_drawing.TestPSDrawing) ... ResourceWarning: unclosed file <_io.TextIOWrapper name='C:\\Users\\rporuri\\AppData\\Local\\Temp\\tmpusjbnlc7\\rendered.eps' mode='w' encoding='cp1252'> [test_ps_drawing.py:17]
ok
test_star_clip (test_ps_drawing.TestPSDrawing) ... ResourceWarning: unclosed file <_io.TextIOWrapper name='C:\\Users\\rporuri\\AppData\\Local\\Temp\\tmpc74amg35\\rendered.eps' mode='w' encoding='cp1252'> [test_ps_drawing.py:17]
ok
test_star_eof_fill (test_ps_drawing.TestPSDrawing) ... ResourceWarning: unclosed file <_io.TextIOWrapper name='C:\\Users\\rporuri\\AppData\\Local\\Temp\\tmp4wgsmrcn\\rendered.eps' mode='w' encoding='cp1252'> [test_ps_drawing.py:17]
ok
test_star_fill (test_ps_drawing.TestPSDrawing) ... ResourceWarning: unclosed file <_io.TextIOWrapper name='C:\\Users\\rporuri\\AppData\\Local\\Temp\\tmprtlf3hns\\rendered.eps' mode='w' encoding='cp1252'> [test_ps_drawing.py:17]
ok
test_text (test_ps_drawing.TestPSDrawing) ... ResourceWarning: unclosed file <_io.TextIOWrapper name='C:\\Users\\rporuri\\AppData\\Local\\Temp\\tmphyf5uun5\\rendered.eps' mode='w' encoding='cp1252'> [test_ps_drawing.py:17]
ok
test_text_clip (test_ps_drawing.TestPSDrawing) ... ResourceWarning: unclosed file <_io.TextIOWrapper name='C:\\Users\\rporuri\\AppData\\Local\\Temp\\tmpbend34qb\\rendered.eps' mode='w' encoding='cp1252'> [test_ps_drawing.py:17]
ok
```

jwiggins reviewed (approved):
LGTM. Note that the GL tests still have this problem:
```
home/travis/.edm/envs/enable-test-3.6-pyqt/lib/python3.6/site-packages/kiva/tests/test_gl_drawing.py:55: ResourceWarning: unclosed file <_io.BufferedWriter name='/tmp/tmp7nfmc1i8/rendered.png'>

  buffer.get_color_buffer().save(filename, encoder=PNGImageEncoder())

ok

test_circle_clip (kiva.tests.test_gl_drawing.TestGLDrawing) ... /home/travis/.edm/envs/enable-test-3.6-pyqt/lib/python3.6/site-packages/kiva/tests/test_gl_drawing.py:55: ResourceWarning: unclosed file <_io.BufferedWriter name='/tmp/tmpg1f78hpl/rendered.png'>

  buffer.get_color_buffer().save(filename, encoder=PNGImageEncoder())

ok

test_circle_fill (kiva.tests.test_gl_drawing.TestGLDrawing) ... /home/travis/.edm/envs/enable-test-3.6-pyqt/lib/python3.6/site-packages/kiva/tests/test_gl_drawing.py:55: ResourceWarning: unclosed file <_io.BufferedWriter name='/tmp/tmpp13utc14/rendered.png'>

  buffer.get_color_buffer().save(filename, encoder=PNGImageEncoder())

ok

test_line (kiva.tests.test_gl_drawing.TestGLDrawing) ... /home/travis/.edm/envs/enable-test-3.6-pyqt/lib/python3.6/site-packages/kiva/tests/test_gl_drawing.py:55: ResourceWarning: unclosed file <_io.BufferedWriter name='/tmp/tmp61yj939u/rendered.png'>

  buffer.get_color_buffer().save(filename, encoder=PNGImageEncoder())

ok

test_quarter_circle (kiva.tests.test_gl_drawing.TestGLDrawing) ... /home/travis/.edm/envs/enable-test-3.6-pyqt/lib/python3.6/site-packages/kiva/tests/test_gl_drawing.py:55: ResourceWarning: unclosed file <_io.BufferedWriter name='/tmp/tmpih3jrfhr/rendered.png'>

  buffer.get_color_buffer().save(filename, encoder=PNGImageEncoder())

ok

test_rect (kiva.tests.test_gl_drawing.TestGLDrawing) ... /home/travis/.edm/envs/enable-test-3.6-pyqt/lib/python3.6/site-packages/kiva/tests/test_gl_drawing.py:55: ResourceWarning: unclosed file <_io.BufferedWriter name='/tmp/tmp0h8fty3k/rendered.png'>

  buffer.get_color_buffer().save(filename, encoder=PNGImageEncoder())

ok

test_rectangle (kiva.tests.test_gl_drawing.TestGLDrawing) ... /home/travis/.edm/envs/enable-test-3.6-pyqt/lib/python3.6/site-packages/kiva/tests/test_gl_drawing.py:55: ResourceWarning: unclosed file <_io.BufferedWriter name='/tmp/tmpnk4i1tug/rendered.png'>

  buffer.get_color_buffer().save(filename, encoder=PNGImageEncoder())

ok

test_star_clip (kiva.tests.test_gl_drawing.TestGLDrawing) ... skipped 'gl graphics context does not support star_clip (#164)'

test_star_eof_fill (kiva.tests.test_gl_drawing.TestGLDrawing) ... /home/travis/.edm/envs/enable-test-3.6-pyqt/lib/python3.6/site-packages/kiva/tests/test_gl_drawing.py:55: ResourceWarning: unclosed file <_io.BufferedWriter name='/tmp/tmpdr6zooxm/rendered.png'>

  buffer.get_color_buffer().save(filename, encoder=PNGImageEncoder())

ok

test_star_fill (kiva.tests.test_gl_drawing.TestGLDrawing) ... /home/travis/.edm/envs/enable-test-3.6-pyqt/lib/python3.6/site-packages/kiva/tests/test_gl_drawing.py:55: ResourceWarning: unclosed file <_io.BufferedWriter name='/tmp/tmpl7q3qpkr/rendered.png'>

  buffer.get_color_buffer().save(filename, encoder=PNGImageEncoder())

ok

test_text (kiva.tests.test_gl_drawing.TestGLDrawing) ... /home/travis/.edm/envs/enable-test-3.6-pyqt/lib/python3.6/site-packages/kiva/tests/test_gl_drawing.py:55: ResourceWarning: unclosed file <_io.BufferedWriter name='/tmp/tmp5l_oqzlg/rendered.png'>

  buffer.get_color_buffer().save(filename, encoder=PNGImageEncoder())
```

rahulporuri commented:
@jwiggins addressed your comment about the resource warnings from `test_gl_drawing` - I didn't notice them earlier because those are skipped on windows. CI is happy and I dont see the resource warnings on travis/linux anymore. Merging.
